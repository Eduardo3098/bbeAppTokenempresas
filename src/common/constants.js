const AppStateValues = {
  ACTIVE: 'active',
};

const AppStateEvents = {
  CHANGE: 'change',
};

const KeyboardEvents = {
  DID_SHOW: 'keyboardDidShow',
  DID_HIDE: 'keyboardDidHide',
};

const LoadStatus = {
  STARTED: 'STARTED',
  DONE: 'DONE',
};

const AccessibilityRole = {
  NONE: 'none',
  BUTTON: 'button',
  LINK: 'link',
  SEARCH: 'search',
  IMAGE: 'image',
  KEYBOARD_KEY: 'keyboardkey',
  TEXT: 'text',
  ADJUSTABLE: 'adjustable',
  IMAGE_BUTTON: 'imagebutton',
  HEADER: 'header',
  SUMMARY: 'summary',
  ALERT: 'alert',
  CHECKBOX: 'checkbox',
  COMBOBOX: 'combobox',
  MENU: 'menu',
  MENUBAR: 'menubar',
  MENUITEM: 'menuitem',
  PROGRESSBAR: 'progressbar',
  RADIO: 'radio',
  RADIO_GROUP: 'radiogroup',
  SCROLLBAR: 'scrollbar',
  SPIN_BUTTON: 'spinbutton',
  SWITCH: 'switch',
  TAB: 'tab',
  TAB_LIST: 'tablist',
  TIMER: 'timer',
  TOOLBAR: 'toolbar',
};

const CHANNEL_AMOUNT_LIMIT = 15000;
const MULTIPLIER_TO_MILLIS = 1000;
const PARSE_DEFAULT_RADIX = 10;
const STRING_EMPTY = '';
const STRING_SPACE = ' ';
const ENCODING_BASE_64 = 'base64';
const DEFAULT_DECIMAL_SEPARATOR = '.';
const REGEX_FLAG_GLOBAL = 'g';
const KEY_MAX_PERCENTAGE = 100;
const CLIPBOARD_CONTENT_MAX_LENGTH = 20;
const PHONE_NUMBER_LENGTH = 8;
const PHONE_NUMBER_MAX_LENGTH = 10;
const PHONE_NUMBER_MASK_LENGTH = 3;
const TOKEN_FULL_PERCENTAGE = 100;
const MAX_CODE_LENGTH = 8;

const REGEX_APP_STATE_INACTIVE_BACKGROUND = /inactive|background/;

export {
  AppStateValues,
  AppStateEvents,
  KeyboardEvents,
  LoadStatus,
  AccessibilityRole,
  CHANNEL_AMOUNT_LIMIT,
  STRING_EMPTY,
  STRING_SPACE,
  ENCODING_BASE_64,
  DEFAULT_DECIMAL_SEPARATOR,
  REGEX_FLAG_GLOBAL,
  PARSE_DEFAULT_RADIX,
  MULTIPLIER_TO_MILLIS,
  KEY_MAX_PERCENTAGE,
  CLIPBOARD_CONTENT_MAX_LENGTH,
  PHONE_NUMBER_LENGTH,
  PHONE_NUMBER_MAX_LENGTH,
  PHONE_NUMBER_MASK_LENGTH,
  REGEX_APP_STATE_INACTIVE_BACKGROUND,
  TOKEN_FULL_PERCENTAGE,
  MAX_CODE_LENGTH,
};
