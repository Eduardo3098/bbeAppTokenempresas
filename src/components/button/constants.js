const BUTTON_COLORS = {
  PRIMARY: 'primary',
  SECONDARY: 'secondary',
};

const BUTTON_SIZE = {
  SMALL: 'small',
  MEDIUM: 'medium',
  LARGE: 'large',
  EXTRA_LARGE: 'extra-large',
};

export {BUTTON_COLORS, BUTTON_SIZE};
