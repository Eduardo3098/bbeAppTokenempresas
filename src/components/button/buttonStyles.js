import {Platform} from 'react-native';
import styled from 'styled-components/native';
import {ColorPalette} from '../../res/colors';
import {FontsList} from '../../res/fonts';
import {pixelNormalizer} from '../../common/fontUtil';
import {BUTTON_COLORS} from './constants';

const getButtonBackgroundColor = ({color, disabled}) => {
  if (color === BUTTON_COLORS.PRIMARY) {
    return disabled ? ColorPalette.primaryDisabled : ColorPalette.primary;
  }
  return ColorPalette.darkGrayishBlue200;
};

const getButtonBorderColor = ({color, disabled}) => {
  if (color === BUTTON_COLORS.PRIMARY) {
    return disabled ? ColorPalette.primaryDisabled : ColorPalette.primary;
  }
  return disabled
    ? ColorPalette.darkGrayishBlue200
    : ColorPalette.darkGrayishBlue;
};

const Container = styled.TouchableOpacity`
  background-color: ${getButtonBackgroundColor};
  border-radius: 6px;
  border-width: 1px;
  border-style: solid;
  border-color: ${getButtonBorderColor};
  width: auto;
  min-width: 120px;
  justify-content: center;
  align-items: center;
  padding: 13px 0;
`;

const getTextColor = ({disabled}) =>
  disabled ? ColorPalette.secondaryDisabled : ColorPalette.secondary;

const Text = styled.Text`
  color: ${getTextColor};
  font-family: ${FontsList.PreloSemiBold};
  font-size: ${pixelNormalizer(14)};
`;

export {Container, Text};
