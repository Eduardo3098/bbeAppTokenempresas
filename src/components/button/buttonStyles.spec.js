import React from 'react';
import renderer from 'react-test-renderer';

import {Container, Text} from './buttonStyles';
import {BUTTON_COLORS} from './constants';

describe('Button components', () => {
  it('should render primary ButtonContainer successfully', () => {
    const component = <Container color={BUTTON_COLORS.PRIMARY} />;
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render primary disabled ButtonContainer successfully', () => {
    const component = <Container color={BUTTON_COLORS.PRIMARY} disabled />;
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render primary with status ButtonContainer successfully', () => {
    const component = <Container status="WAIT" color={BUTTON_COLORS.PRIMARY} />;
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render secondary ButtonContainer successfully', () => {
    const component = <Container color={BUTTON_COLORS.SECONDARY} />;
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render secondary disabled ButtonContainer successfully', () => {
    const component = <Container color={BUTTON_COLORS.SECONDARY} disabled />;
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render ButtonText successfully', () => {
    const component = <Text />;
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render ButtonText disabled successfully', () => {
    const component = <Text disabled />;
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
