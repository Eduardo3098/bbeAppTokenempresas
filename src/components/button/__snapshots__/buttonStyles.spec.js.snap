// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button components should render ButtonText disabled successfully 1`] = `
<Text
  disabled={true}
  style={
    Array [
      Object {
        "color": "#9fa8be",
        "fontFamily": "Prelo-SemiBold",
        "fontSize": 28,
      },
    ]
  }
/>
`;

exports[`Button components should render ButtonText successfully 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#0F265C",
        "fontFamily": "Prelo-SemiBold",
        "fontSize": 28,
      },
    ]
  }
/>
`;

exports[`Button components should render primary ButtonContainer successfully 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFDD00",
      "borderColor": "#FFDD00",
      "borderRadius": 6,
      "borderStyle": "solid",
      "borderWidth": 1,
      "justifyContent": "center",
      "minWidth": 120,
      "opacity": 1,
      "paddingBottom": 10,
      "paddingLeft": 0,
      "paddingRight": 0,
      "paddingTop": 18,
      "width": "auto",
    }
  }
/>
`;

exports[`Button components should render primary disabled ButtonContainer successfully 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFF38D",
      "borderColor": "#FFF38D",
      "borderRadius": 6,
      "borderStyle": "solid",
      "borderWidth": 1,
      "justifyContent": "center",
      "minWidth": 120,
      "opacity": 1,
      "paddingBottom": 10,
      "paddingLeft": 0,
      "paddingRight": 0,
      "paddingTop": 18,
      "width": "auto",
    }
  }
/>
`;

exports[`Button components should render primary with status ButtonContainer successfully 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFDD00",
      "borderColor": "#FFDD00",
      "borderRadius": 6,
      "borderStyle": "solid",
      "borderWidth": 1,
      "justifyContent": "center",
      "minWidth": 120,
      "opacity": 1,
      "paddingBottom": 10,
      "paddingLeft": 0,
      "paddingRight": 0,
      "paddingTop": 18,
      "width": "auto",
    }
  }
/>
`;

exports[`Button components should render secondary ButtonContainer successfully 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#e9ecf3",
      "borderColor": "#D4D8E1",
      "borderRadius": 6,
      "borderStyle": "solid",
      "borderWidth": 1,
      "justifyContent": "center",
      "minWidth": 120,
      "opacity": 1,
      "paddingBottom": 10,
      "paddingLeft": 0,
      "paddingRight": 0,
      "paddingTop": 18,
      "width": "auto",
    }
  }
/>
`;

exports[`Button components should render secondary disabled ButtonContainer successfully 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#e9ecf3",
      "borderColor": "#e9ecf3",
      "borderRadius": 6,
      "borderStyle": "solid",
      "borderWidth": 1,
      "justifyContent": "center",
      "minWidth": 120,
      "opacity": 1,
      "paddingBottom": 10,
      "paddingLeft": 0,
      "paddingRight": 0,
      "paddingTop": 18,
      "width": "auto",
    }
  }
/>
`;
